### PR TITLE
implemented py-cache-decorator

### PR DESCRIPTION
**Pull request by mariiahorbova**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+    @wraps(func)
+    def inner(*args) -> Callable:
+        if args in cached:
+            print("Getting from cache")
+            return cached[args]
+        else:
+            result = func(*args)
+            cached[args] = result
+            print("Calculating new result")
+            return result
```
Comment (LiudmylaKulinchenko) [inline app/main.py:17]: Try to rewrite your decorator not to have `return` duplicated

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
-    pass
+from typing import Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    cached = {}
+
+    @wraps(func)
+    def inner(*args) -> Callable:
```
Comment (LiudmylaKulinchenko) [inline app/main.py:9]: Does your function really returns Callable?

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (mariiahorbova) [reply app/main.py]: done

File: app/main.py (hunk 1 above)
Comment (mariiahorbova) [reply app/main.py]: done

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+
+    @wraps(func)
+    def inner(*args) -> Any:
+        if args in cached:
+            print("Getting from cache")
+            return cached[args]
+        else:
+            cached[args] = func(*args)
+            print("Calculating new result")
+            return cached[args]
```
Comment (LiudmylaKulinchenko) [inline app/main.py:16]: You can use a single return statement out of the if/else statement instead of two inside it ☀

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 3 above)
Comment (mariiahorbova) [reply app/main.py]: ☀

**Review by YuriiKindrat — APPROVED** [review]

**Pull request closed without merging**